tests/test_database.py

```diff
@@ -71,12 +71,13 @@ def test_add_workout(self):
         # Adding new user
         database.add_user(new_user)
 
+        database.get_user_class_by_username(new_user.username)
```

Comment:
You fetched the database user but never assigned it back to new_user? 

`db_user = database.get_user_class_by_username(new_user.username)`

Then remove line 86 since you're doing that here and don't need to refetch.